<?xml version="1.0"?>
<launch>
    <!-- 保持与原有接口完全兼容 -->
    <arg name="rviz" default="false" />
    <arg name="map" default="/home/ch/catkin_ws/src/algorithm/FASTLIO2_SAM_LC/pcd/suo_scans.pcd" />
    <arg name="map_hz" default="5"/>
    <arg name="map_frame" default="map" />
    <arg name="map_topic" default="map_env"/>  <!-- 重要：保持为map -->
    
    <!-- 新增的优化参数（保持向后兼容） -->
    <arg name="use_latched_publisher" default="true" />
    <arg name="map_voxel" default="0.0" />
    <arg name="fast_format" default="true" />
    
    <!-- Ultra Fast Map Publisher (替换原pcl_ros节点) -->
    <node pkg="ultra_fast_map_publisher" type="ultra_fast_map_publisher" 
          name="map_publisher" output="screen">
        <!-- 核心参数 -->
        <param name="map_path" value="$(arg map)" />
        <param name="output_topic" value="$(arg map_topic)" />  <!-- 使用原话题名 -->
        <param name="frame_id" value="$(arg map_frame)" />
        
        <!-- 原参数映射 -->
        <param name="latch_mode" value="$(arg use_latched_publisher)" />
        <param name="publish_rate" value="$(arg map_hz)" />
        
        <!-- 新增优化功能 -->
        <param name="use_fast_format" value="$(arg fast_format)" />
        <param name="auto_convert" value="true" />
        <param name="voxel_size" value="$(arg map_voxel)" />
        
        <!-- 性能优化 -->
        <param name="num_threads" value="0" />  <!-- 自动检测 -->
        <param name="memory_mapping" value="true" />
    </node>
    
    <!-- 保持原有RViz配置 -->
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" 
              args="-d $(find mapping)/launch/rviz/3d_map_load.rviz" />
    </group>
</launch>